packages/custom-elements/src/core.ts

```diff
@@ -58,15 +58,34 @@ export function createSignal<T>(id: string, initialValue: T): Signal<T> {
 	return signal(initialValue);
 }
 
+type Resolver = (value: Signal<any>) => void;
+
 class SignalStore {
 	store = new Map<string, Signal<any>>();
-	resolvers = new Map<string, (value: any) => void>();
+	resolvers = new Map<string, Array<Resolver>>();
+
+	private setResolver(id: string, value: (v: Signal<any>) => void) {
+		if (!this.resolvers.has(id)) {
+			this.resolvers.set(id, []);
+		}
+
+		this.resolvers.get(id)!.push(value);
+	}
+
+	private async resolveAllFor(id: string, signal: Signal<any>): Promise<void> {
+		await Promise.all(
+			(this.resolvers.get(id) ?? []).map((fn) => {
+				fn(signal);
+			}),
+		);
+	}
```

Comment:
@nmn I went ahead and implemented the logic for `this.resolvers`. I believe this is what you had in mind?

Reply:
I split it out into #10